src/passes/LLVMNontrappingFPToIntLowering.cpp

```diff
@@ -0,0 +1,150 @@
+#include "pass.h"
+#include "wasm-builder.h"
+#include "wasm.h"
+#include <limits>
+
+// By default LLVM emits nontrapping float-to-int instructions to implement its
+// fptoui/fptosi conversion instructions. This pass replaces these instructions
+// with code sequences which also implement LLVM's fptoui/fptosi, but which are
+// not semantically equivalent in wasm. This is because out-of-range inputs to
+// these instructions produce poison values. So we need only ensure that there
+// is no trap, but need not ensure any particular result.
```

Comment:
Can you maybe give a concrete example here in the top comment? Reading this, I'm not sure where the poison values appear (in the form this receives, or the form this outputs). And maybe a summary of the possible behavior changes after the pass.